src/Language/PureScript/Ide/Filter/Declaration.hs

```diff
@@ -0,0 +1,57 @@
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE DeriveFoldable  #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+
+module Language.PureScript.Ide.Filter.Declaration
+       ( IdeDeclaration(..)
+       , DeclarationType(..)
+       , typeDeclarationForDeclaration
+       ) where
+
+import           Protolude                     hiding (isPrefixOf)
+
+import           Data.Aeson
+import qualified Language.PureScript.Ide.Types as PI
+
+data DeclarationType = Value
+  | Type
+  | Synonym
+  | DataConstructor
+  | TypeClass
+  | ValueOperator
+  | TypeOperator
+  | Kind
+  deriving (Show, Eq, Ord)
+
+instance FromJSON DeclarationType where
+  parseJSON (String s) = case s of
```

Comment:
This can be written with aeson's `withText` function to give a better parse error.

`parseJson = withText "Declaration type tag" $ \t -> case t of`